integrationtests/test_i18n.py

```diff
@@ -20,3 +53,88 @@ def test_js_english_interpolation(self):
+        )
+
+    def test_language_switch_in_django_templates(self):
+        # You can change from "en" to "el" and back and the UI will be translated
+        b = self.browser
+        b.visit("/")
+        b.assert_element('input[name="password"][placeholder="Password"]', wait=True)
+        b.assert_element('button[type="submit"]', text="Sign In")
+        self.switch_locale_django("el")
+        b.assert_element('input[name="password"][placeholder="Συνθηματικό"]', wait=True)
+        b.assert_element('button[type="submit"]', text="Σύνδεση")
+        self.switch_locale_django("en")
+        b.assert_element('input[name="password"][placeholder="Password"]', wait=True)
+        b.assert_element('button[type="submit"]', text="Sign In")
+
+    def test_language_switch_in_react(self):
+        # You can change from "en" to "el" and back and the UI will be translated
+        b = self.browser
+
+        accounts.login(b, self.user.email, self.user.email)
+        b.visit("/workflows/")
+
+        b.assert_element(
+            'form.create-workflow button[type="submit"]',
+            text="Create Workflow",
+            wait=True,
+        )
+        self.switch_locale_react("en", "el")
+        b.assert_element(
+            'form.create-workflow button[type="submit"]',
+            text="Δημιουργία ροής εργασιών",
+            wait=True,
+        )
+        self.switch_locale_react("el", "en")
+        b.assert_element(
+            'form.create-workflow button[type="submit"]',
+            text="Create Workflow",
+            wait=True,
+        )
+
+
+class TestLocaleOnLogin(LocaleSwitcherIntegrationTest):
+    def setUp(self):
+        super().setUp()
+        self.user = self.account_admin.create_user(
+            "user@example.org", first_name="Jane", last_name="Doe"
+        )
+
+    def test_remember_locale_on_login(self):
+        # When a user logs out, their locale continues to be used
+        # When a user logs in, their previous locale is used.
+        b = self.browser
+
+        # Login (in English)
+        accounts.login(b, self.user.email, self.user.email)
+        b.visit("/workflows/")
+
+        b.assert_element(
+            'form.create-workflow button[type="submit"]',
+            text="Create Workflow",
+            wait=True,
+        )
+
+        # Switch to Greek and then logout. Page must still be in Greek
+        self.switch_locale_react("en", "el")
+        accounts.logout(b, "el")
+        #
+        b.assert_element('input[name="password"][placeholder="Συνθηματικό"]', wait=True)
+        b.assert_element('button[type="submit"]', text="Σύνδεση")
+
+        # Switch to English and then login. After login, page must be in Greek
+        self.switch_locale_django("en")
+        accounts.login(b, self.user.email, self.user.email, "en", "el")
+        b.assert_element(
+            'form.create-workflow button[type="submit"]',
+            text="Δημιουργία ροής εργασιών",
+            wait=True,
+        )
```

Comment:
I'm not usually a fan of a test that tests three things ... but I'm going to recommend a fourth here :).

What language does the user get after logout? (IIRC we agreed it should be Greek.)

Reply:
Logout is tested above; as mentioned in the comment, `Switch to Greek and then logout. Page must still be in Greek`.

Reply:
I was talking about testing something different. What I'm asking to test is:

1. Create account that has Greek
2. Browse in English
3. Log in [should now be Greek] (does something we haven't integration-tested)
4. Log out [should now be Greek] <-- this is the test I'm requesting

This is different from what's already in the test, which is:

1. Create account
2. Log in
3. Set to Greek (we know, through integration tests, that this sets both COOKIE _and_ user-profile language)
4. Log out [should now be Greek] <-- this is what you've already tested

Since this confused me -- and our conversation wasn't seamless -- could you please create this new test separate from the existing one that tests three things?

Reply:
Let's see if I understand correctly.

Current test:
1. Create account
2. Log in
3. Set to Greek. Notice that now I have created a user with Greek locale in preferences.
4. Logout (must be in Greek)
5. Change to English
6. Login (must be in Greek)
7. Logout (must be in Greek)

I should split it in two
Test logout:
1. Create account
2. Log in
3. Set to Greek.
4. Logout (must be in Greek)

Test login:
1. Create account with Greek locale in preferences
2. Login from English page (must go to Greek page)
3. Logout (must be in Greek)

Reply:
First, please leave the current test as-is and add a new test.

My point is, the sequence of events I asked to test is subtly different from the sequence that is already tested. This matters because elsewhere in my review of this pull request I'm suggesting you stop writing a cookie :). I worry that not-writing the cookie will not cause a failure in your test but may cause a failure in the test I'm proposing.